spec/java/util/Random.lsl

```diff
@@ -0,0 +1,228 @@
+        doubles (Random),
+        doubles (Random, double, double),
+        doubles (Random, long),
+        doubles (Random, long, double, double),
+
+        ints (Random),
+        ints (Random, int, int),
+        ints (Random, long),
+        ints (Random, long, int, int),
+
+        longs (Random),
+        longs (Random, long),
+        longs (Random, long, long),
+        longs (Random, long, long, long),
+
+        nextBoolean,
+        nextBytes,
+        nextDouble,
+        nextFloat,
+        nextGaussian,
+        nextInt (Random),
+        nextInt (Random, int),
+        nextLong (),
+
+        setSeed,
+    ];
+
+
+    // constructors
+
+    constructor Random (@target self: Random)
+    {
+        action TODO();
+    }
+
+
+    constructor Random (@target self: Random, arg0: long)
+    {
+        action TODO();
+    }
+
+
+    // utilities
+
+    // static methods
+
+    // methods
+
+    fun doubles (@target self: Random): DoubleStream
+    {
+        action TODO();
+    }
+
+
+    fun doubles (@target self: Random, arg0: double, arg1: double): DoubleStream
+    {
+        action TODO();
+    }
+
+
+    fun doubles (@target self: Random, arg0: long): DoubleStream
+    {
+        action TODO();
+    }
+
+
+    fun doubles (@target self: Random, arg0: long, arg1: double, arg2: double): DoubleStream
+    {
+        action TODO();
+    }
+
+
+    fun ints (@target self: Random): IntStream
+    {
+        action TODO();
+    }
+
+
+    fun ints (@target self: Random, arg0: int, arg1: int): IntStream
+    {
+        action TODO();
+    }
+
+
+    fun ints (@target self: Random, arg0: long): IntStream
+    {
+        action TODO();
+    }
+
+
+    fun ints (@target self: Random, arg0: long, arg1: int, arg2: int): IntStream
+    {
+        action TODO();
+    }
+
+
+    fun longs (@target self: Random): LongStream
+    {
+        action TODO();
+    }
+
+
+    fun longs (@target self: Random, arg0: long): LongStream
+    {
+        action TODO();
+    }
+
+
+    fun longs (@target self: Random, arg0: long, arg1: long): LongStream
+    {
+        action TODO();
+    }
+
+
+    fun longs (@target self: Random, arg0: long, arg1: long, arg2: long): LongStream
+    {
+        action TODO();
+    }
+
+
+    fun nextBoolean (@target self: Random): boolean
+    {
+        result = action SYMBOLIC("boolean");
+    }
+
+
+    fun nextBytes (@target self: Random, arg0: array<byte>): void
+    {
+        action TODO();
+    }
+
+
+    fun nextDouble (@target self: Random): double
+    {
+        result = action SYMBOLIC("double");
+        action ASSUME(0.0d <= result && result < 1.0d);
+    }
+
+
+    fun nextFloat (@target self: Random): float
+    {
+        result = action SYMBOLIC("float");
+        action ASSUME(0.0f <= result && result < 1.0f);
+    }
+
+
+    @synchronized fun nextGaussian (@target self: Random): double
+    {
+        result = action SYMBOLIC("double");
+        action ASSUME(-1.0d <= result && result < 1.0d);
```

Comment:
I have run such code: "println(Random().nextGaussian())" 
It returns me "-1.3354423265964728"

Reply:
Fixed